sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/highlighting/internal/DefaultHighlighting.java

```diff
@@ -47,8 +48,7 @@ public DefaultHighlighting(@Nullable SensorStorage storage) {
     super(storage);
     syntaxHighlightingRuleSet = Sets.newTreeSet(new Ordering<SyntaxHighlightingRule>() {
       @Override
-      public int compare(@Nullable SyntaxHighlightingRule left,
-        @Nullable SyntaxHighlightingRule right) {
+      public int compare(@Nonnull SyntaxHighlightingRule left, @Nonnull SyntaxHighlightingRule right) {
```

Comment:
Contract is inherited and constraints can't be relaxed. It would break Liskov substitution principle. I had the same pb with Guava Function. The solution was to create class NonNullInputFunction (see in sonar-core)